### PR TITLE
feat: added catalogId and displayPath support

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✔                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✖                                                                       |

Add tiles path field to catalog record

### CODE REVIEW

**Review by cl-shaharshavit-rnd — APPROVED** [review]

File: src/yaml/rasterCatalog/insertRequest.yaml [hunk 1]
```diff
@@ -10,11 +10,16 @@ components:
       required:
         - metadata
         - links
+        - id
       allOf:
         - $ref: "./updateRequest.yaml#/components/schemas/UpdateRequest"
         - properties:
+            id:
```
Comment (cl-shaharshavit-rnd) [inline src/yaml/rasterCatalog/insertRequest.yaml:17]: remove this as the id is already part of the metadata

File: src/yaml/rasterCatalog/updateRequest.yaml [hunk 2]
```diff
@@ -14,4 +14,7 @@ components:
           type: array
           items:
             $ref: ../link.yaml#/components/schemas/Link
-      additionalProperties: false
+
```
Comment (cl-shaharshavit-rnd) [inline src/yaml/rasterCatalog/updateRequest.yaml:17]: is it still nessary?

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

File: src/models/rasterCatalog/rasterCatalogUpsertRequest.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import { Link } from '../common';
 import { LayerMetadata } from '../layerMetadata';
 
 export interface IRasterCatalogUpsertRequestBody {
+  id: string;
```
Comment (asafMasa) [inline src/models/rasterCatalog/rasterCatalogUpsertRequest.ts:5]: same as @cl-shaharshavit-rnd remark, isn't it already exists in metadata?

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/models/pycsw/interfaces/pycswCoreModel.ts [hunk 4]
```diff
@@ -2,7 +2,6 @@ import { Link } from '../../layerMetadata/link';
 
 export interface IPycswCoreModel {
   // PYSCW CORE FIELDS
-  id: string | undefined;
```
Comment (cl-shaharshavit-rnd) [inline src/models/pycsw/interfaces/pycswCoreModel.ts:5]: revert

File: src/models/layerMetadata/layerMetadata.ts [hunk 5]
```diff
@@ -17,6 +17,7 @@ import { getTsTypesMapping, ITsTypesMapping, tsTypes, TsTypes } from './decorato
 import { ProductType } from './enums';
 
 export interface ILayerMetadata {
+  id: string | undefined;
```
Comment (cl-shaharshavit-rnd) [inline src/models/layerMetadata/layerMetadata.ts:20]: ask client team

**Review by cl-shaharshavit-rnd — CHANGES_REQUESTED** [review]

**Review by cl-shaharshavit-rnd — APPROVED** [review]

File: src/models/layerMetadata/layerMetadata.ts [hunk 6]
```diff
@@ -37,6 +37,32 @@ export interface IPropPYCSWMapping extends IPYCSWMapping {
...
+      primary: true,
+    },
+  })
+  @tsTypes({
+    mappingType: TsTypes.STRING,
+  })
+  @graphql()
+  @fieldConfig({
+    category: FieldCategory.MAIN,
+    isAutoGenerated: true,
```
Comment (asafMasa) [inline src/models/layerMetadata/layerMetadata.ts:61]: it's not auto generated

Ask @alebinson if some property should be added so it won't be displayed

**Review by asafMasa — CHANGES_REQUESTED** [review]

**Review by asafMasa — APPROVED** [review]

**Review by alebinson — APPROVED** [review]
In general PR is Ok.
There is an issue with allowed for editing fields the number of those fields is very limited (only 5 fields must be editable due to mc-models.xls)

I will open a relevant issue for raster team

**Pull request merged**